### PR TITLE
Build Docker image with grakn-core

### LINKED ISSUE #4889: Create a Docker image distribution
- Bazel target: `//:distribution-docker`
- image name: `grakn-core`
- tag: Grakn Core's current version number (eg., 1.5.0)
- OS: https://github.com/phusion/baseimage-docker

Grakn consists of two processes - `Grakn` and `GraknStorage`. Be sure to research on the caveat of running multiple processes on a single Docker container, particularly about cleaning up in order to make sure not to leave zombie process on the users' machine.

### PR DESCRIPTION
**Pull request by vmax**
## What is the goal of this PR?

Fixes #4889 
Allows us to build Docker image with `grakn-core` with `bazel`

## What are the changes implemented in this PR?

- adds reference to `@io_bazel_rules_docker`
- imports [`openjdk/8`](https://hub.docker.com/_/openjdk) as base Docker image
- adds `//:distribution-docker` target
- adds `grakn_docker.sh` to serve as Docker entrypoint

### CODE REVIEW

File: grakn_docker.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
```
Comment (haikalpribadi) [inline grakn_docker.sh]: Can we rename this file to use hyphen rather than underscore, @vmax ?

**Comment by haikalpribadi** [comment]
where is `@bazelbuild_rules_docker`? Do you mean `io_bazel_rules_docker`?

**Comment by haikalpribadi** [comment]
Another one: shouldn't we build docker image in CI too @vmax ?

**Comment by vmax** [comment]
@haikalpribadi it is built by 'build everything' command [here](https://github.com/graknlabs/grakn/blob/master/.circleci/config.yml#L58)

**Pull request merged**